sql/src/main/java/io/crate/operation/fetch/PositionalBucketMerger.java

```diff
@@ -143,30 +140,42 @@ private void findLeastBucketIt() {
 
     private boolean emitRow(Row row) {
         outputCursor++;
-        return downstream.setNextRow(row);
+        return multiUpstreamRowDownstream.setNextRow(row);
     }
 
 
     public void downstream(RowDownstream downstream) {
-        this.downstream = downstream.registerUpstream(this);
+        multiUpstreamRowDownstream.downstreamHandle(downstream.registerUpstream(this));
     }
 
     public PositionalBucketMerger registerUpstream(RowUpstream upstream) {
-        upstreamsRemaining.incrementAndGet();
+        multiUpstreamRowDownstream.registerUpstream(upstream);
         return this;
     }
 
     public void finish() {
-        if (upstreamsRemaining.decrementAndGet() <= 0) {
-            downstream.finish();
-        }
+        multiUpstreamRowDownstream.finish();
     }
 
     public void fail(Throwable throwable) {
         consumeBuckets.set(false);
-        downstream.fail(throwable);
+        multiUpstreamRowDownstream.fail(throwable);
+    }
+
+    @Override
+    public void pause() {
+        multiUpstreamRowUpstream.pause();
     }
 
+    @Override
+    public void resume() {
+        multiUpstreamRowUpstream.resume();
+    }
+
+    @Override
+    public boolean setNextRow(Row row) {
+        throw new UnsupportedOperationException("setNextRow not supported, use setNextBucket");
```

Comment:
does the PositionalBucketMerger need to implement the Handle interface if it doesn't support it ?